lib/Core/Executor.cpp

```diff
@@ -3785,11 +3788,46 @@ void Executor::executeMemoryOperation(ExecutionState &state,
                                       bool isWrite,
                                       ref<Expr> address,
                                       ref<Expr> value /* undef if read */,
-                                      KInstruction *target /* undef if write */) {
+                                      KInstruction *target /* undef if write */,
+                                      bool isNontemporal /* false if read */) {
   Expr::Width type = (isWrite ? value->getWidth() :
                      getWidthForLLVMType(target->inst->getType()));
   unsigned bytes = Expr::getMinBytesForWidth(type);
 
+  // To reduce duplicated code between fast and slow cases
+  auto doOperation =
+    [&](ExecutionState &state,
+        const MemoryObject *mo,
+        const ObjectState *os,
+        ref<Expr> offset) {
+
+      if (isWrite) {
+        if (os->readOnly) {
+          terminateStateOnError(state, "memory error: object read only",
+                                ReadOnly);
+        } else {
+          ObjectState *wos = state.addressSpace.getWriteable(mo, os);
+          wos->write(state, offset, value);
```

Comment:
Are we issuing warnings if this write updates a cache line that is already dirty? It seems like there could be a case where the programmer wrote
```
store ... => X
<other instructions>
store ... => X !nontemporal
sfence
```

but they meant

```
store ... => X
clwb X
sfence
<other instructions>
store ... => X !nontemporal
sfence
```

Under our model, we cannot known for a fact that this is a bug, but it might be nice to issue a warning letting the programmer know this is happening.

It would also be nice to know from our perspective, because I'm curious if there's any code where that would occur.

Reply:
Well, as you've said before, we're focusing on persistence bugs, not semantic ones.

Furthermore, there would be no distinction between the example you gave and the following:
```
store ... => cl[0] !nontemporal
store ... => cl[1] !nontemporal
...
store ... => cl[63] !nontemporal
sfence
```

And I'd much rather not output anything for cases like that than output something for cases like yours.

Reply:
TODO pending time.